pkg/ebpf/ingress_node_firewall_loader.go

```diff
@@ -238,19 +244,25 @@ func (infc *IngNodeFwController) IngressNodeFwAttach(ifacesName []string, isDele
 	return ifMap, nil
 }
 
-// cleanup will delete all link objects for all interfaces and remove all the maps
+// cleanup will delete interface's eBPF objects.
 func (infc *IngNodeFwController) cleanup(ifName string) error {
 	l, ok := infc.links[ifName]
 	if !ok {
 		return fmt.Errorf("failed to find Link object for interface %s", ifName)
 	}
+	if err := l.Unpin(); err != nil {
+		return fmt.Errorf("failed to unpin link for %s err: %q", ifName, err)
+	}
 	if err := l.Close(); err != nil {
-		return err
+		return fmt.Errorf("failed to close and detach link %s err: %q", ifName, err)
 	}
 	delete(infc.links, ifName)
 	if len(infc.links) == 0 {
 		if err := infc.objs.Close(); err != nil {
```

Comment:
That close here creates an issue. When I attach all interfaces, then detach all interfaces -> Close()
A subsequent attach will then fail with an invalid file descriptor because the objs are closed. There's no way for a user of an instance of infc to know if the object was closed, and it doesn't reinitialise closed objects by itself.

Reply:
so we should just never close objs nor pinDir and if daemon crashes we create new instance and go from there

Reply:
as discussed on slack new api to cleanup when ds go down